File: test-sources/typescript/player/player.ts
import { PLAYER_TYPES } from './player-types';

export class Player {
    name: string;
    power: number;

    type = PLAYER_TYPES.SOLDIER;

    constructor(name: string) {
        this.name = name;
    }

    init(power: number) {
        this.power = power;
    }
}

File: test-sources/java/src/main/java/player/fr/sii/game/player/Player.java
package main.java.player.fr.sii.game.player;

public class Player {

    private String name;
    private int power;
    private PlayerType type;

    public Player(String pName) {
        this.name = pName;
        this.type = PlayerType.SOLDIER;
    }

    public void init(int pPower){
        this.power = pPower;
    }
}
